tool/src/webapp/js/forum-summary.js

```diff
@@ -0,0 +1,70 @@
+var itemsToShow;
+$(function() {
+	//escape markup
+	String.prototype.escapeHTML = function(){
+		return (this.replace(/&/g, '&amp;').replace(/>/g, '&gt;').replace(/</g, '&lt;').replace(/"/g, '&quot;'));
+	};
+	itemsToShow = $('.forum-summary-div').find($('#numberOfConversations')).val();
+	//set the 'numberofconversations' in the hidden field for edit screen
+	$('#numberOfMessages').val(itemsToShow);
+	showForums();
+});
+
+function showForums(){
+	var forumsUrl = $('.forum-summary-site-url').text().replace(/'/g,"");
+	var messagesArray = [];
+	var errorText;
+	//Get forums
+	$.ajaxSetup({ cache: false,async: false });	//Disable caching for ajax requests and set to synchronous (because of nesting)
+	$.getJSON(forumsUrl+'.json', function(data_forum_collection) {
+		$(data_forum_collection["forums_collection"]).each(function(fc_index, fc_value){
+			$.getJSON(forumsUrl+'/forum/'+fc_value["entityId"]+'.json', function(data_forum) {
+				$(data_forum["topics"]).each(function(f_index,f_value){
+					$.getJSON(forumsUrl+'/forum/'+fc_value["entityId"]+'/topic/'+f_value['id']+'.json', function(data_topic) {
+						$(data_topic["threads"]).each(function(t_index,t_value){
+							$.getJSON(forumsUrl+'/forum/'+fc_value["entityId"]+'/topic/'+f_value['id']+'/message/'+t_value['messageId']+'.json', function(data_message) {
```

Comment:
This may have poor scaling as you end up with one AJAX request for each thread, in a site with 100 threads that's allot of requests.